tests/unit/TestAlignment.cpp

```diff
@@ -274,6 +274,24 @@ TEST(LinearAlignmentTests, BasicTest)
     EXPECT_EQ(14, score);
     delete a;
 
+    a = AlignLinear(
+        "CGAC",
+        "GAAAACGAC",
+        &score);
```

Comment:
Nice small test case here.  This test is just checking for a crash ... could you check the result as well, here?  I won't require that for the longer sequences below.